lib/Elastica/Reindex.php

```diff
@@ -0,0 +1,117 @@
+<?php
+namespace Elastica;
+
+use Elastica\Query\AbstractQuery;
+
+class Reindex
+{
+    const VERSION_TYPE = 'version_type';
+    const VERSION_TYPE_INTERNAL = 'internal';
+    const VERSION_TYPE_EXTERNAL = 'external';
+    const OPERATION_TYPE = 'op_type';
+    const OPERATION_TYPE_CREATE = 'create';
+    const CONFLICTS = 'conflicts';
+    const CONFLICTS_PROCEED = 'proceed';
+    const TYPE = 'type';
+    const SIZE = 'size';
+    const QUERY = 'query';
+
+    public static function reindex(Index $oldIndex, Index $newIndex, array $options = [])
```

Comment:
I was thinking passing params to constructor and then having something like `$reindex->Run()` or `Start` or something similar. Perhaps Start() could also take additional options. Not 100% sure yet how the most common use case will look like. Any thoughts?